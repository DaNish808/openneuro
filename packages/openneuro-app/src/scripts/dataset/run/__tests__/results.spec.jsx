import React from 'react'
import { shallow } from 'enzyme'
import Results from '../results.jsx'

describe('dataset/run/Results', () => {
  it('should render something when the job status is "SUCCEEDED"', () => {
    const successfulRun = {
      results: [
        {
          type: 'file',
          dirPath: 'avg_brain_size.txt',
          name: 'avg_brain_size.txt',
          path:
            'openneuro.outputs/cf02b5bf17df80d056effadccffd65bd/dcedf0c3-4438-466d-9f85-61c75275b355/avg_brain_size.txt',
        },
      ],
      analysis: {
        status: 'SUCCEEDED',
      },
    }
    expect(
      shallow(
        <Results
          run={successfulRun}
          acknowledgements="acknowledge"
          displayFile={jest.fn()}
          toggleFolder={jest.fn()}
        />,
      ),
    ).toMatchSnapshot()
  })
  it('should render something when the job status is "FAILED"', () => {
    const failedRun = {
      results: [
        {
          type: 'file',
          dirPath: 'avg_brain_size.txt',
          name: 'avg_brain_size.txt',
          path:
            'openneuro.outputs/cf02b5bf17df80d056effadccffd65bd/dcedf0c3-4438-466d-9f85-61c75275b355/avg_brain_size.txt',
        },
      ],
      analysis: {
        status: 'FAILED',
      },
    }
    expect(
      shallow(
        <Results
          run={failedRun}
          acknowledgements="acknowledge"
          displayFile={jest.fn()}
          toggleFolder={jest.fn()}
        />,
      ),
    ).toMatchSnapshot()
  })
  it('should render nothing when job status is neither "SUCCEEDED" nor "FAILED"', () => {
    const noResultsRun = {
      results: [],
      analysis: {
        status: 'RUNNING',
      },
    }
    expect(
      shallow(
        <Results
          run={noResultsRun}
          acknowledgements="acknowledge"
          displayFile={jest.fn()}
          toggleFolder={jest.fn()}
        />,
      ).getElement(),
    ).toBe(null)
  })
})
