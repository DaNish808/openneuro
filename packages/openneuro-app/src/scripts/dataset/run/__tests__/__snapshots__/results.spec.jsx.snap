// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataset/run/Results should render something when the job status is "FAILED" 1`] = `
<Accordion
  accordion={true}
  className="results"
>
  <Panel
    bsClass="panel"
    bsStyle="default"
    className="fade-in"
    defaultExpanded={false}
    header="results"
  >
    <div
      className="app-acknowledgements"
    >
      <label>
        Acknowledgements
      </label>
      <div
        className="markdown"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>acknowledge</p>
",
          }
        }
      />
    </div>
    <hr />
    <DownloadAll
      run={
        Object {
          "analysis": Object {
            "status": "FAILED",
          },
          "results": Array [
            Object {
              "dirPath": "avg_brain_size.txt",
              "name": "avg_brain_size.txt",
              "path": "openneuro.outputs/cf02b5bf17df80d056effadccffd65bd/dcedf0c3-4438-466d-9f85-61c75275b355/avg_brain_size.txt",
              "type": "file",
            },
          ],
        }
      }
    />
    <div
      className="file-structure fade-in panel-group"
    >
      <div
        className="panel panel-default"
      >
        <div
          aria-expanded="false"
          className="panel-collapse"
        >
          <div
            className="panel-body"
          >
            <FileTree
              displayFile={[Function]}
              editable={false}
              getFileDownloadTicket={[Function]}
              toggleFolder={[MockFunction]}
              tree={
                Array [
                  Object {
                    "dirPath": "avg_brain_size.txt",
                    "name": "avg_brain_size.txt",
                    "path": "openneuro.outputs/cf02b5bf17df80d056effadccffd65bd/dcedf0c3-4438-466d-9f85-61c75275b355/avg_brain_size.txt",
                    "type": "file",
                  },
                ]
              }
            />
          </div>
        </div>
      </div>
    </div>
  </Panel>
</Accordion>
`;

exports[`dataset/run/Results should render something when the job status is "SUCCEEDED" 1`] = `
<Accordion
  accordion={true}
  className="results"
>
  <Panel
    bsClass="panel"
    bsStyle="default"
    className="fade-in"
    defaultExpanded={false}
    header="results"
  >
    <div
      className="app-acknowledgements"
    >
      <label>
        Acknowledgements
      </label>
      <div
        className="markdown"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>acknowledge</p>
",
          }
        }
      />
    </div>
    <hr />
    <DownloadAll
      run={
        Object {
          "analysis": Object {
            "status": "SUCCEEDED",
          },
          "results": Array [
            Object {
              "dirPath": "avg_brain_size.txt",
              "name": "avg_brain_size.txt",
              "path": "openneuro.outputs/cf02b5bf17df80d056effadccffd65bd/dcedf0c3-4438-466d-9f85-61c75275b355/avg_brain_size.txt",
              "type": "file",
            },
          ],
        }
      }
    />
    <div
      className="file-structure fade-in panel-group"
    >
      <div
        className="panel panel-default"
      >
        <div
          aria-expanded="false"
          className="panel-collapse"
        >
          <div
            className="panel-body"
          >
            <FileTree
              displayFile={[Function]}
              editable={false}
              getFileDownloadTicket={[Function]}
              toggleFolder={[MockFunction]}
              tree={
                Array [
                  Object {
                    "dirPath": "avg_brain_size.txt",
                    "name": "avg_brain_size.txt",
                    "path": "openneuro.outputs/cf02b5bf17df80d056effadccffd65bd/dcedf0c3-4438-466d-9f85-61c75275b355/avg_brain_size.txt",
                    "type": "file",
                  },
                ]
              }
            />
          </div>
        </div>
      </div>
    </div>
  </Panel>
</Accordion>
`;
